docs/mathematics/next-permutation.mdx

```diff
@@ -0,0 +1,128 @@
+
+Implementation of next permutation, it rearranges numbers into the lexicographically next greater permutation of numbers.
+If such an arrangement is not possible, it must rearrange it as the lowest possible order (sorted in ascending order).
+
+## Explanation 
+We need to find the next lexicographic permutation of the given list of numbers than the number formed by the given array.
+Our approach would be inplace.
+
+
+
+Let's have few examples: 
+```
+Input: nums = [1,2,3]
+Output: [1,3,2]
+
+
+Input: nums = [3,2,1]
+Output: [1,2,3]
+
+
+Input: nums = [1]
+Output: [1]
+
+```
+
+## Algorithm
+
+* We will find length of array and store it in `length` variable.
+
+* We will check a base condition if the `length` is less than or equal to 2 simply return `reverse` of array ( as that's the next permutation). 
+
+* Now, Let's keep a pointer at `length - 2` position of array and call it `pointer`.
+
+* Now, we will iterate with conditions `pointer >=0` and Element at that index must be graeter than or equal to Element at next index simply `array[pointer] >= array[pointer + 1]`
+  Here we will be decrementing pointer as `pointer = pointer -1` (ensures us that we are moving backwards over the array)
+
+* Now, Over that process once the `pointer` is equal to `-1` then it simply means we have to return `array.reverse()` 
+
+* If it is not yet equal to `-1' then we iterate over the range of `length - 1, pointer` with incrementer as `-1` and variable as `number`.
+
+* Inside the loop we check condition if `array[pointer] < array[number]` is `True` then swap `array[pointer]` with `array[number]` and `break` after that is done.
+
+* Finally we index over our array from `pointer + 1` till `end` of array and simply reverse is using `reversed` function.
+
+* Now, we'd simply return the modified array as our answer.
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="Python"
+  values={[
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="py">
+
+```python 
+
+
+# Function for next Permutation
+def nextPermutation(array):
+
+ 
+        length = len(array)
+
+        if length <= 2:
+            return array.reverse()
+        pointer = length - 2
+        
+        while pointer >= 0 and array[pointer] >= array[pointer + 1]:
+            pointer -= 1
+        
+        if pointer == -1:
+            return array.reverse()
+        
+        for number in range(length - 1, pointer, -1):
+            if array[pointer] < array[number]:
+                array[pointer], array[number] = array[number], array[pointer]
+                break
+        
+        array[pointer + 1:] = reversed(array[pointer + 1:])
+
+        return array
+
+
+if __name__ == "__main__":
+    #User input for both the strings
+    print('Enter values of the array')
+    array = list(map(int, input().rstrip().split()))
+    print('Next Permutation is: ',nextPermutation(array))  
+
+```
+</TabItem>
+
+</Tabs>

```

Comment:
The Code does not even render on the Website. Did you do a proper check on your local machine before pushing the PR?

Reply:
But it works fine in the deployed version @HarshCasper. Can you please confirm?


![Screenshot (286)](https://user-images.githubusercontent.com/65017645/116231851-06aed500-a777-11eb-993e-e21cd6ccdf2d.png)